### DISCUSSION TITLE
Considering current pop-up design...

### DISCUSSION THREAD

**Discussion by ShibuyaCyana** [musescore/MuseScore#20432] [Polls]
Pop-ups, uhmmmm...
Didn't notice this with the harp update, but now it comes to guitar capo and tuning, and later it will come to [text](https://github.com/musescore/MuseScore/pull/18436). I assume in a future version, a single click on everything will just bring up a popup, and the property panel, once being regarded as a good design, will ultimately become useless and disappear. Wow...

Useless grumblings:
>I don't like popups. I'd like to use another word for this, but someone told me it's too strong, so I'll just say that. It blocks my viewing and editing of content, force me to focus on something else, like filling in an endless list of parameters. This kind of interrupt is almost a punishment. If a popup appears unexpectedly, then most likely it will be bad news (S***lius crashed). Many software offers trial version that uses popup to annoy user, persuading them to buy a license. Long long ago Office suite was using popup to ask user to fill in shape properties, now they use a panel on the right, much like MU's property panel; I'm getting so used to it, that when I was trying a free alternative, right clicking on a shape and select "Edit Style"  - Oh no it's that giant property popup...the devil is back, blocking everything and not letting me to leave, until I finally decided to confirm or cancel him...

So, problems? Here they are:
1. We already have a good property panel out there;
2. If we must use a popup, then please don't make it appear on a single left click...
a. If it is the first object of multi selection (ctrl+click), a popup will appear, blocking me from selecting anything under it until I select something else;
b. dragging...I have to warn myself constantly, that if an item has popup, then I can't drag it with my first click:
![property](https://github.com/musescore/MuseScore/assets/39703247/1375e3a3-b5fe-437d-99b8-1a4ccba7c08a)

I've heard popups are introduced due to user not opening the property panel then can't find things in it. I'd like everything to be put into the property panel, but I guess we can't go back now. So, copying design from Office and we get this:
![image](https://github.com/musescore/MuseScore/assets/39703247/58720da2-e84f-4852-b819-e3d579e226f4)

I assume everyone with any experience level of using Office will be able to find things like this, right?

**Comment by cbjeukendrup** [comment]
I don't really have an opinion about this, but the obvious argument in favour of popups is that you can stay more focussed: instead of constantly switching between the score and the properties panel, you can just keep your eyes focussed on the score.

Also: 
- the interaction between popups and dragging elements will still be improved over time.
- it is not true that all functionality from the properties panel will be replaced by popups. 
I think that Properties panel will contain properties of an element once it exists, while popups will contain controls to set "what the element is" / the "content" of the element (yes, that's vague, I know...)

I don't think making these popups right-click only is a good idea, because then we're hiding the most crucial functionality (namely setting the "content" (e.g. which harp pedals, or which capo, etc.) to right-click only which is _very_ undiscoverable.

**Comment by MarcSabatella** [comment]
No doubt there is room for refinement in exactly how popups work, but in general, I agree with @cbjeukendrup that hiding things in right-click menus is exactly the sort of non-discoverability that we've been trying to move *away* from.  So to me that's a non-starter.

I didn't answer the poll because none of the choices fits.  The "correct" response would be, "Use them where it makes sense to aid both discoverability and usability".

**Comment by ShibuyaCyana** [comment]
(rewrite the poll as it was only my own complains but not gathering opinions)
Some other patches, as added into the poll:
1. double click, as we use that to edit text;
2. an option to merge popup with property panel, or a "popup panel". This might be best of both worlds but seems need a bunch of code works.

Off topic but somewhat related: sometimes I move a note accidentally, later find the spacing become messed up but not sure which note is causing the problem. In this case the old "double click to edit" design works better, but it's too clumsy if I only need to move a hairpin.

**Comment by MarcSabatella** [comment]
Went with option 2, but it's still a pretty biased poll without a clear statement of what I believe.  That is, it's not about whether "dragging / multi selection behavior needs a fix" but more generally, "there is room for further refinement".

**Reply by ShibuyaCyana** [reply]
I should have said "things like blahblah"...
Anyway I'll keep it there or it will be cleared again

**Comment by Tantacrul** [comment]
I think the main point being raised in the discussion makes a few assumptions.

1. Imperfections with popups won't be fixed.
2. We're planning on getting rid of the Properties panel.
3. We're going to blindly use popups for everything.

There's no basis for any of this.

The general idea for popups is to use them in cases where a user doesn't need to interact with it all the time. It'd be a very bad idea if we used popups when you select an accidental for example. However, it's quite a good idea for capo settings, which you are not going to be interacting with hundreds of times in a session. As for the Harp diagram popup? I just completed a new piece where I used it constantly. It was a godsend. The only issue is that you sometimes can't move the item because the popup interferes with the movement. That's just a bug to be fixed.

One point I should raise about the current approach that I think needs addressing: our intention when we introduced this concept was that we'd always **also** to have popup settings duplicated in the Properties panel. @bkunda. We've not really done this, which I think might store up a problem for us down the line.

In some cases, like text styling, we want to have it available in both places: the properties panel and as a popup. I'd point out that MS3 had something kind of like this. It was the right idea but not the best realisation. We're just restoring the idea with an ideal realisation that matches how numerous other text editors have worked for around 15 years.

The general rule is to only use a pattern where it is useful. We don't need to use the same pattern for everything. So chill out.

**Reply by bkunda** [reply]
Yes I think we need to allocate some time going into v.4.4 to duplicating the existing popup controls in Properties, particularly before we start contemplating any further popups.

**Reply by avvvvve** [reply]
Created an issue for the above here! #21083
